test/test_auth.py

```diff
@@ -220,6 +221,65 @@ def delegate_get_credentials():
             credentials = credentials_future.result(TIMEOUT)
 
 
+@unittest.skipUnless(os.environ.get('AWS_TESTING_COGNITO_IDENTITY'),
+                     'set env var to run test: AWS_TESTING_COGNITO_IDENTITY')
+class CognitoCredentialsProviderTest(NativeResourceTest):
+
+    def test_unauthenticated(self):
+        identity = os.environ.get('AWS_TESTING_COGNITO_IDENTITY')
+        tls_opts = awscrt.io.TlsContextOptions()
+        tls_context = awscrt.io.ClientTlsContext(tls_opts)
+
+        provider = awscrt.auth.AwsCredentialsProvider.new_cognito(
+            endpoint="cognito-identity.us-east-1.amazonaws.com",
+            identity=identity,
+            tls_ctx=tls_context
+        )
+
+        self.assertIsNotNone(provider)
+        credentials = provider.get_credentials().result(TIMEOUT)
+        self.assertIsNotNone(credentials)
+
+    def test_maximal_create(self):
+        identity = os.environ.get('AWS_TESTING_COGNITO_IDENTITY')
+        tls_opts = awscrt.io.TlsContextOptions()
+        tls_context = awscrt.io.ClientTlsContext(tls_opts)
+
+        provider = awscrt.auth.AwsCredentialsProvider.new_cognito(
+            endpoint="cognito-identity.us-east-1.amazonaws.com",
+            identity=identity,
+            tls_ctx=tls_context,
+            logins=[('provider1', 'token1'), ('provide2', 'token2')],
+            custom_role_arn='not-a-real-arn'
+        )
+
+        self.assertIsNotNone(provider)
+
+
+@unittest.skipUnless(os.environ.get('AWS_TESTING_COGNITO_IDENTITY') and os.environ.get('AWS_TEST_HTTP_PROXY_HOST')
+                     and os.environ.get('AWS_TEST_HTTP_PROXY_PORT'), 'set env var to run test: AWS_TESTING_COGNITO_IDENTITY')
+class CognitoCredentialsProviderProxyTest(NativeResourceTest):
+
+    def test_unauthenticated(self):
+        identity = os.environ.get('AWS_TESTING_COGNITO_IDENTITY')
+        tls_opts = awscrt.io.TlsContextOptions()
+        tls_context = awscrt.io.ClientTlsContext(tls_opts)
+        http_proxy_options = awscrt.http.HttpProxyOptions(
+            os.environ.get('AWS_TEST_HTTP_PROXY_HOST'),
+            int(os.environ.get('AWS_TEST_HTTP_PROXY_PORT', '0')))
+
```

Comment:
trivial: the `skipUnless()` says PROXY_PORT is required, but then down here you have it defaulting to "0"

Reply:
Well that's just so that it has a valid numerical value if someone does something dumb like set it to 'A'  The test will fail in that case which seems ok?